lib/schema.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+module.exports = {
+  projects: {
+    url: '/accounts/{accountId}/projects',
+    requiredFields: ['accountId'],
+    list: {
+      type: 'GET'
+    },
+    create: {
+      type: 'POST',
+      requiredFields: ['name', 'sitePrimary']
+    },
+    read: {
+      url: '/{projectId}',
+      type: 'GET',
+      requiredFields: ['projectId']
+    },    
+    update: {
+      url: '/{projectId}',
+      type: 'PATCH',
+      requiredFields: ['projectId']
+    },    
+    delete: {
+      url: '/{projectId}',
+      type: 'DELETE',
+      requiredFields: ['projectId']
+    }
+  },
+  crawls: {
+    url: '/accounts/{accountId}/projects/{projectId}/crawls',
+    requiredFields: ['accountId', 'projectId'],
+    list: {
+      type: 'GET'
+    },
```

Comment:
Another nice shortcut to consider would be checking these first to see if they are string, then defaulting the value to being the verb with defaults in that case. If non-string then go through the existing logic. It's minor, but would allow simpler schema code like `list: 'GET'` here